frappe/email/smtp.py

```diff
@@ -169,9 +169,10 @@ def __init__(self, login=None, password=None, server=None, port=None, use_tls=No
 	def setup_email_account(self, append_to=None, sender=None):
 		self.email_account = get_outgoing_email_account(raise_exception_not_set=False, append_to=append_to, sender=sender)
 		if self.email_account:
+			account_password = self.email_account.password
 			self.server = self.email_account.smtp_server
 			self.login = getattr(self.email_account, "login_id", None) or self.email_account.email_id
-			self.password = self.email_account.password
+			self.password = frappe.safe_encode(account_password, 'ascii')
```

Comment:
why encode to ascii! it should be utf-8 right?

Reply:
Hi, perhaps I didn't explain thoroughly. 

The problem is that in some servers configuration smtplib will only work using ascii and not utf-8. In some cases as with our servers using PLESK + POSTFIX combination it won't work with utf-8.  It's a bug in smtplib in python 2 that's noted in a lot of github issues across many python platforms and projects. 
The ideal solution is supposedly to upgrade to python 3 (haven't tested) or just encode the password to ascii. Otherwise those servers won't work authenticate properly causing an exception to be thrown. 

If you have sufficient knowledge you could attempt to verify first if the server will work with unicode and then send as unicode or else encode to ascii but since python is not my field of expertise it would be impossible for me to implement. 

As a matter of fact I tried frappe.safe_encode to utf-8 as well before submitting the commit as ascii  and it failed because the result was still a unicode variable and the error was exactly the same as before. Encoding to ascii works flawlessly in cases where the password won't contain unicode characters.

Reply:
Create a config in Email Account (Use Ascii Password). Why force ASCII on everyone! This will surely break for non ascii users

Reply:
Well essentially safe_encode seems that it would force conversion to ASCII for users that their password does not contain unicode characters. The rest users should still receive a unicode result for their password. However I have no knowledge of whether smtplib will correctly use it in other server configurations that I can't test. I tested it with PLESK + Postfix and Gmail and it works but I can't test the rest. 

```python 
def safe_encode(param, encoding = 'utf-8'):
	try:
		param = param.encode(encoding)
	except Exception:
		pass
	return param
```
In any case where can I find some developer documentation on how to add a checkbox field to an existing doctype programmatically so I can add it as an additional option in Email Account class and revamp this?

Reply:
safe_encode would force utf-8 conversion, not ascii! its a big difference

check the frappe tutorial for learning how to edit doctypes https://frappe.io/docs/user/en/tutorial